cypress/integration/05_collection_spec.js

```diff
@@ -1,164 +1,332 @@
+        .should('have.value', 'test_col')
+        .clear()
+        .type('test_col2')
+      // check properties table 
+      cy.get('.dialogContent')
+        .find('.keys > tr')
+        .should('have.length', 7)
+        .should('contain', 'Created')
+        .should('contain', 'Owner')
+        .should('contain', 'Group')
+      // check permissions table  
+      cy.get('.dialogContent')
+        .find('.permissions-editor > tr')
+        .should('have.length', 3)
+        .should('contain', 'user')
+        .should('contain', 'group')
+        .should('contain', 'other')
+      cy.get('.main')
+        .click()
+      cy.get('.dialogBlock')
+        .should('not.exist');
+      cy.get('[node-id$=test_col2]')
+        .should('exist')
+      cy.get('[node-id$=test_col]')
+        .should('not.exist')
+    })
+
+    it('should not create duplicate collection', () => {
+      cy.get('[node-id$=test]')
+        .rightclick()
+        .then(() => {
+          cy.get('.p-Menu')
+            .should('be.visible')
+            .contains('New collection')
+            .trigger('mousemove')
+          cy.get('[data-command="fusion.new-collection"]')
+            .should('be.visible')
+            .click()
+          cy.focused()
+            .clear()
+            .type('test_col2{enter}')
+          cy.get('.error')
+            .should('exist')
+            .should('contain.text', 'Item already exists')
+        })
+    })
+
+    it('should create nested collection', () => {
+      cy.get('[node-id$=test_col2]')
+        .click()
+        .rightclick()
+      cy.get('.p-Menu')
+        .should('be.visible')
+        .contains('New collection')
+      cy.get('[data-command="fusion.new-collection"]')
+        .should('be.visible')
+        .click()
+      cy.focused()
+        .clear()
+        .type('test_colA{enter}')
+      // TODO(DP): we migh want to check the proper nesting more explicitely,
+      // but that is already covered by checking for this collection after deleting
+      // its parent collection 
+      cy.get('.fusion-view')
+        .contains('test_colA')
+        .should('exist')
+    })
 
-      it('should let users rename collection', () => {
-        cy.get('[node-id$=untitled-1]')
-          .rightclick()
-        cy.get('[data-command="fusion.rename"]')
-          .should('be.visible')
-          .contains('Rename')
-          .click()
-        cy.focused()
-          .type('test_col{enter}')
-        cy.get('@fetch').should('be.calledWithMatch', Cypress.env('API_HOST') + '/exist/restxq/fusiondb/collection?uri=/db/test_col', {
-          method: 'PUT',
-          headers: { 'x-fs-move-source': '/db/untitled-1' },
-        });
-        cy.get('.fusion-view')
-          .contains('test_col')
-        cy.get('[node-id$=untitled-1]')
-          .should('not.exist')
+    it('should move a collection', () => {
+      const dataTransfer = new DataTransfer();
+      cy.get('[node-id$="test\\/col1"]')
```

Comment:
another good place to reuse existing test or system artifacts instead of creating our own.

Reply:
We need to create `dataTransfer` object for two tests only, so I thought it would be overkill to create it under `beforeAll`, and we shouldn't create in `before` because it shouldn't be reused.

Reply:
my point wasn't about the `before` hooks but about the collections which have been created from earlier test cases, lets re-use whats already been generated and tested.